packages/eslint-plugin/src/configs/recommended.json

```diff
@@ -60,7 +39,190 @@
     "@angular-eslint/no-output-on-prefix": "error",
     "@angular-eslint/no-output-rename": "error",
     "@angular-eslint/no-outputs-metadata-property": "error",
-    "@angular-eslint/use-lifecycle-interface": "warn",
-    "@angular-eslint/use-pipe-transform-interface": "error"
+    "@angular-eslint/use-lifecycle-interface": "error",
+    "@angular-eslint/use-pipe-transform-interface": "error",
+    "@typescript-eslint/adjacent-overload-signatures": "error",
+    "@typescript-eslint/array-type": "off",
+    "@typescript-eslint/ban-types": [
+      "error",
+      {
+        "types": {
+          "Object": {
+            "message": "Avoid using the `Object` type. Did you mean `object`?"
+          },
+          "Function": {
+            "message": "Avoid using the `Function` type. Prefer a specific function type, like `() => void`."
+          },
+          "Boolean": {
+            "message": "Avoid using the `Boolean` type. Did you mean `boolean`?"
+          },
+          "Number": {
+            "message": "Avoid using the `Number` type. Did you mean `number`?"
+          },
+          "String": {
+            "message": "Avoid using the `String` type. Did you mean `string`?"
+          },
+          "Symbol": {
+            "message": "Avoid using the `Symbol` type. Did you mean `symbol`?"
+          }
+        }
+      }
+    ],
+    "@typescript-eslint/consistent-type-assertions": "error",
+    "@typescript-eslint/dot-notation": "error",
+    "@typescript-eslint/member-delimiter-style": [
```

Comment:
`member-delimiter-style` is covered by Prettier, I think, as well as a few others -- e.g. `space-before-function-paren`. Is that intended?

Reply:
The initial work is about being as close to what the Angular CLI ships OOTB today. The Angular CLI does not use prettier.

For sure I would encourage anyone I speak to personally to use prettier, and I will also trend this config away from stylistic rules completely over time, but for now parity is the main objective

Reply:
@JoshuaKGoldberg FYI in a follow up PR I have increased the clarity around compatibility with Angular CLI's TSLint setup vs "recommended" and also split out formatting related rules into their own named config (and added in a note in the manual config example I have in the integration tests): https://github.com/angular-eslint/angular-eslint/blob/456c1667bd20cef61f42966959ca8e58169c9769/packages/integration-tests/fixtures/v1014-multi-project-manual-config/.eslintrc.json#L47